classes/notification/managerDelegate/EditingStatusNotificationManager.inc.php

```diff
@@ -0,0 +1,218 @@
+				return $dispatcher->url($request, ROUTE_PAGE, $context->getPath(), 'workflow', 'access', $notification->getAssocId());
+		}
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getStyleClass()
+	 */
+	public function getStyleClass($notification) {
+		return NOTIFICATION_STYLE_CLASS_INFORMATION;
+	}
+
+	/**
+	 * @copydoc NotificationManagerDelegate::updateNotification()
+	 */
+	public function updateNotification($request, $userIds, $assocType, $assocId) {
+		$context = $request->getContext();
+		$contextId = $context->getId();
+
+		assert($assocType == ASSOC_TYPE_SUBMISSION);
+		$submissionId = $assocId;
+		$submissionDao = Application::getSubmissionDAO();
+		$submission = $submissionDao->getById($submissionId);
+
+		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+		$editorStageAssignments = $stageAssignmentDao->getEditorsAssignedToStage($submissionId, $submission->getStageId());
+
+		// One way to check if a copyeditor is assigned
+		// Get the 'copyedit assignment' notifications
+		$copyEditorAssigned = false;
+		$notificationDao = DAORegistry::getDAO('NotificationDAO'); /* @var $notificationDao NotificationDAO */
+		$copyEditAssignmentNotificationFactory = $notificationDao->getByAssoc(
+			ASSOC_TYPE_SUBMISSION,
+			$submissionId,
+			null,
+			NOTIFICATION_TYPE_COPYEDIT_ASSIGNMENT,
+			$contextId
+		);
+		// If there are 'copyedit assignment' notifications
+		if (!$copyEditAssignmentNotificationFactory->wasEmpty()) {
+			$copyEditAssignmentNotifications = $copyEditAssignmentNotificationFactory->toArray();
+			// Check if at least one user is still in the participants list i.e. assigned
+			foreach ($copyEditAssignmentNotifications as $copyEditAssignmentNotification) {
+				$copyEditorStageAssignmentFactory = $stageAssignmentDao->getBySubmissionAndStageId($submissionId, WORKFLOW_STAGE_ID_EDITING, null, $copyEditAssignmentNotification->getUserId());
+				if (!$copyEditorStageAssignmentFactory->wasEmpty()) {
+					$copyEditorAssigned = true;
+					break;
+				}
+			}
+		}
+
+		// Another way to check if a copyeditor is assigned
+		// Get the copyediting discussions
+		$queryDao = DAORegistry::getDAO('QueryDAO'); /* @var $queryDao QueryDAO */
+		$queries = $queryDao->getByAssoc(ASSOC_TYPE_SUBMISSION, $submissionId, WORKFLOW_STAGE_ID_EDITING);
+
+		// Get the copyedited files
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO');
+		import('lib.pkp.classes.submission.SubmissionFile');
+		$copyeditedFiles = $submissionFileDao->getLatestRevisions($submissionId, SUBMISSION_FILE_COPYEDIT);
+
+		$notificationType = $this->getNotificationType();
+
+		foreach ($editorStageAssignments as $editorStageAssignment) {
+			switch ($submission->getStageId()) {
+				case WORKFLOW_STAGE_ID_PRODUCTION:
+					// Remove 'assign a copyeditor' and 'awaiting copyedits' notification
+					$this->_removeNotification($submissionId, $editorStageAssignment->getUserId(), $notificationType, $contextId);
+					break;
+				default:
```

Comment:
Better, I think, to specify `case WORKFLOW_STAGE_ID_EDITING` and add a `default: assert(false);` for other cases.